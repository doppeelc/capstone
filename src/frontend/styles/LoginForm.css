.LoginForm {
    width: 13%;
    margin: auto;
}