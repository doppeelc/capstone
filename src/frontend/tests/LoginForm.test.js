import { render } from '@testing-library/react';
import LoginForm from '../components/LoginForm';

test('renders without crashing', () => {
    render(<LoginForm />);
});

test('matches the snapshot', () => {
    const { asFragment } = render(<LoginForm />);
    expect(asFragment()).toMatchSnapshot();
});

test('renders the form empty', () => {
    const { getByPlaceholderText } = render(<LoginForm />);
    expect(getByPlaceholderText("username")).toBeInTheDocument();
    expect(getByPlaceholderText("username")).toBeEmptyDOMElement();
    expect(getByPlaceholderText("password")).toBeInTheDocument();
    expect(getByPlaceholderText("password")).toBeEmptyDOMElement();
});






