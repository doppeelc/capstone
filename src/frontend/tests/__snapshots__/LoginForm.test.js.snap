// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <h1>
      Log In
    </h1>
    <form
      id="LoginForm"
    >
      <label
        for="username"
        id="lbl"
      >
        Username
      </label>
      <input
        id="inpt"
        name="username"
        placeholder="username"
        value=""
      />
      <label
        for="password"
        id="lbl"
      >
        Password
      </label>
      <input
        id="inpt"
        name="password"
        placeholder="password"
        value=""
      />
      <button>
        Log in
      </button>
    </form>
  </div>
</DocumentFragment>
`;
